core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/ConnectionUtils.java

```diff
@@ -47,4 +63,27 @@ public HttpURLConnection connectToEndpoint(URI endpoint, Map<String, String> hea
         return connection;
     }
 
+    private int resolveMetadataServiceTimeoutMillis() {
+
+        String timeoutValue = SystemSettingUtils.resolveSetting(SdkSystemSetting.AWS_METADATA_SERVICE_TIMEOUT)
```

Comment:
I agree with this observation and attempted to move the timeout setting out of the util class. However, ConnectionUtils is shared by InstanceProfileCredentialsProvider, EC2MetadataUtils, and indirectly by AutoDefaultsModeDiscovery and InstanceProfileRegionProvider.

Additionally, the util package already includes IMDS-specific classes, such as Ec2MetadataConfigProvider. Adding this logic to ConnectionUtils helps avoid modifying SdkProtected classes, where overloading methods to pass timeouts could result in non-overloaded methods not using the configured timeouts. Given the current design, with ConnectionUtils shared across these components, it seems like the most practical place to add this logic during instantiation.

Do you think we should proceed with adding overloaded methods to SdkProtected APIs, or is there a better alternative we could consider?

Reply:
I see what you mean, my concern is that if `ConnectionUtils` is used for accessing non-imds endpoints, those http calls will use the timeouts configured for imds, it might be confusing for users. Should that be a concern? Do you think there is another better way to avoid that?

Reply:
It seems `ConnectionUtils` is also used by `ContainerCredentialsProvider`, so we may want to avoid applying the same env var to `ContainerCredentialsProvider`

Reply:
Done